### ISSUE TITLE
error, can't open files "cannot create context"

### ISSUE THREAD

**Issue by sebastinas** [pwmt/zathura-pdf-mupdf#46]
*On GitLab by @gaoDean on Mar 15, 2023, 06:54*

---

Whenever I open a pdf document, it comes up with the following error:

```
cannot create context: incompatible header (1.20.0) and library (1.21.1) versions
error: could not open document
```

zathura version:
```
zathura 0.5.2
girara 0.3.9 (runtime: 0.3.9)
(plugin) pdf-mupdf (0.3.9) (/usr/local/Cellar/zathura/HEAD-81e541e/lib/zathura/libpdf-mupdf.dylib)
```

**Comment by sebastinas** [comment]
*On GitLab by @kamsh on Mar 22, 2023, 15:53*

---

I was getting the following error whenever I tried to open any pdf with zathura:

```
cannot create context: incompatible header (1.21.1) and library (1.19.0) versions
error: could not open document
```

# My System

I couldn't find anywhere in the zathura-pdf-mupdf source code where the version is hard coded so I am not sure what is causing this problem.

```
$pacman -Q zathura zathura-pdf-mupdf mupdf libmupdf girara
zathura 0.5.2-1
zathura-pdf-mupdf 0.4.0-1
mupdf 1.21.1-1
libmupdf 1.21.1-1
girara 0.3.9-1

$ zathura --version
zathura 0.5.2
girara 0.3.9 (runtime: 0.3.9)
(plugin) pdf-mupdf (0.4.0) (/usr/lib/zathura/libpdf-mupdf.so)
```
# Possible Fix

I have an arch flavor distribution and I guess I had accidently self-compiled and installed libmupdf which led to the creation of a file `/usr/lib/libmupdf.so`

Eventually I updated my system and it ended up updating `/usr/lib/libmupdf-third.a` and `/usr/lib/libmupdf.a` but not the dynamic library. So when I used AUR to compile and install zathura-pdf-mupdf, I guess parts of it would link to `/usr/lib/libmupdf.so` (version 1.19.0), but the rest of the stuff was getting updated using the static libraries (the *.a libs) which were version 1.21.0.

I fixed this by `rm /usr/lib/libmupdf.so` and recompilling zathura-pdf-mupdf. Now it runs, although it crashed whenver I try to open the table of contents, but I believe thats a zathura problem.

# TLDR
Step 1. Check to see if you have `/usr/bin/libmupdf.so`. If you do then delete it.
Step 2. Recompile zathura-pdf-mupdf

**Comment by sebastinas** [comment]
Nothing that we can do on our side. mupdf does not usually come with a shared library. Whoever provides that to you, needs to proper SONAME transitions on mupdf version bumps.
